src/control/jobs/film_jobs.c

```diff
@@ -253,12 +254,15 @@ static void dt_film_import1(dt_job_t *job, dt_film_t *film)
     g_free(cdn);
 
     /* import image */
-    dt_image_import(cfr->id, (const gchar *)image->data, FALSE);
+    const int imgid = dt_image_import(cfr->id, (const gchar *)image->data, FALSE);
 
     fraction += 1.0 / total;
     dt_control_job_set_progress(job, fraction);
 
-
+    if((imgid & 7) == 7)
```

Comment:
I would even go with 3. The CPU will mostly be spent in updating the thumbs preview and having 7 to do at the same time but less frequently wont change the fact that thumb preview computing is heavy.

Fact is that recent work from @AlicVB to have a way to use jpeg preview only for a certain size of thumbs will be a good solution here.